### PR TITLE
Fix an issue caused by unexpected stop of trace sessions when TraceEventSession is just created and attached to check if a session is active.

### PR DESCRIPTION
**Pull request by ZhaminXu**
Fix an issue caused by unexpected stop of trace sessions when TraceEventSession is just created and attached to check if a session is active.

### CODE REVIEW

**Comment by ZhaminXu** [comment]
We encountered an issue when trying to use latest TraceEvent nuget package. Trace sessions are closed unexpected due to GC-ed TraceEventSession objects. So propose this fix, as current logic will caused trouble to users whoever uses GetActionSession method right after creating the real trace session.

**Comment by brianrob** [comment]
@ZhaminXu, thanks for the contribution.  Do you know which line(s) in ```Dispose()``` are causing the attached session to be closed?  I'm concerned that we might leak resources if ```Dispose``` doesn't get called at all even in the attach case.

**Comment by ZhaminXu** [comment]
> @ZhaminXu, thanks for the contribution.  Do you know which line(s) in Dispose() are causing the attached session to be closed?  I'm concerned that we might leak resources if Dispose doesn't get called at all even in the attach case.
Yes, it's Stop(true) method. I also checked more code there. If TraceEventSession is created and attached, m_sessionHandle is invliad, m_source and m_kernelSession are not used. So I think it's safe to skip the entire dispose method. But after thinking carefully, I think your concern is valid. let me try to just skip the stop method.

File: src/TraceEvent/TraceEventSession.cs [hunk 1]
```diff
@@ -867,7 +867,14 @@ public void Dispose()
                 if (m_StopOnDispose)
                 {
                     m_StopOnDispose = false;
-                    Stop(true);
+
+                    // Only dispose session and source when required.
+                    // For session just attached to check if it's active, we must not call dispose.
+                    // Otherwise, it will caused unexpected stop of trace sessions.
+                    if (m_Create)
```
Comment (brianrob) [inline src/TraceEvent/TraceEventSession.cs]: Can you update the comment here to state that we only stop the session when we were the original creator of it and not for cases where we attach?  It looks like this is the original comment, but it's no longer correct.  Thanks.

Comment (brianrob) [reply src/TraceEvent/TraceEventSession.cs]: Otherwise, this looks good to me.  Thanks!

Comment (ZhaminXu) [reply src/TraceEvent/TraceEventSession.cs]: done. Thanks for reviewing!

**Pull request merged**